src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1556,6 +1556,11 @@ internal void ReportFieldContextualKeywordConflictIfAny(ParameterSyntax syntax,
         /// <returns>true if managed type-related errors were found, otherwise false.</returns>
         internal static bool CheckManagedAddr(CSharpCompilation compilation, TypeSymbol type, Location location, BindingDiagnosticBag diagnostics, bool errorForManaged = false)
         {
+            // Skip the check for error types that represent truly missing types (not found),
+            // but still report for error types due to other issues (e.g., inaccessibility).
+            if (type is ErrorTypeSymbol { ResultKind: LookupResultKind.Empty or LookupResultKind.NotATypeOrNamespace })
```

Comment:
>LookupResultKind.Empty or LookupResultKind.NotATypeOrNamespace [](http://example.com/codeflow?start=54&length=62)

Are there tests that observe significance of this condition? I.e. they would fail if the condition was removed.

Reply:
Added a couple of demonstrative tests in 7242579cea6